src/components/Header/Header.test.js

```diff
@@ -1,18 +1,69 @@
 import React from 'react';
+import { Provider } from 'react-redux';
 import renderer from 'react-test-renderer';
+import configureStore from 'redux-mock-store';
 
-import Header from './Header';
+import { Header } from './Header';
+import { initialState } from 'reducers/application';
 import { withMemoryRouter } from 'utils';
 // Prevent findDOMNode error in test from Dropdown component in react-bootstrap
 jest.mock('react-dom', () => ({
   findDOMNode: () => ({})
 }));
 
 describe('<Header />', () => {
-  it('renders correctly', () => {
-    const RoutedHeader = withMemoryRouter(Header);
-    const component = renderer.create(<RoutedHeader />);
+  const actions = {
+    logoutUser: jest.fn()
+  };
+  const mockStore = configureStore();
+  const store = mockStore({ application: initialState });
+  const RoutedHeader = withMemoryRouter(Header);
 
-    expect(component.toJSON()).toMatchSnapshot();
+  it('renders logged out correctly', () => {
+    const props = {
+      loggedIn: false
+    };
+
+    expect(
+      renderer
+        .create(
+          <Provider store={store}>
+            <RoutedHeader actions={actions} {...props} />
```

Comment:
If this is the inner, unconnected Header export, then there should not be a need for a `<Provider>` - if `Header` itself uses/references a connected component, that can be mocked like

```
jest.mock('components/SomeComponent/SomeComponent', () =>
  require('utils').mockComponent('SomeComponent')
);
```